File: cpp/se/PostalInfoType.cpp
#include "se/PostalInfoType.hpp"

// Static member initialisation.
std::vector<const EnumType *> PostalInfoType::values;

const PostalInfoType* PostalInfoType::INTERNATIONAL()
{
	static const PostalInfoType type("int");
	return &type;
}

const PostalInfoType* PostalInfoType::LOCAL()
{
	static const PostalInfoType type("loc");
	return &type;
}

PostalInfoType::PostalInfoType(const std::string &type)
    : EnumType (values, type)
{ }

void PostalInfoType::init()
{
	INTERNATIONAL();
	LOCAL();
}

const PostalInfoType* PostalInfoType::value (const std::string &name)
{
    return (const PostalInfoType *)EnumType::value (name, values);
}

File: src/main/java/com/ausregistry/jtoolkit2/se/PostalInfoType.java
package com.ausregistry.jtoolkit2.se;

/**
 * Enumeration of PostalInfo types supported by EPP.  The only difference
 * between the two types is the allowed character encoding; international
 * postal info is restricted to US ASCII, whereas local postal info element
 * content may be represented in unrestricted UTF8.
 */
public enum PostalInfoType {
    INTERNATIONAL("int"),
    LOCAL("loc");

    private String type;

    PostalInfoType(String type) {
        this.type = type;
    }

    public String toString() {
        return type;
    }
}

